### DISCUSSION TITLE
How to change modulation order?

### DISCUSSION THREAD

**Discussion by SKZnljx** [Telecominfraproject/oopt-gnpy#483] [Q&A]
Hi,
I would like to ask where I can set the modulation order.
For C+L band fixed-grid transmission, the idle wavelength will have a dummy light, which is unmodulated. Your previous reply mentioned that GNPy assumes "full channel load" by default, is this load modulated? If so, can it be changed to unmodulated?
One more question, where should I set the modulation of the transmission service?
Thanks.

**Comment by EstherLerouzic** [comment]
Dear @SKZnljx 
you may find part of the answer in this thread: https://github.com/Telecominfraproject/oopt-gnpy/discussions/434
The current engine only cares about baud_rate and signal roll-off. Modulation format impact is only modelled through the minimum required OSNR at receiver.

Now regarding your question, gnpy-path-request script assume full load with all channels being identical on each path for the computed services.
gnpy-transmission-main script  allows the user to define a mixed type of channels thanks to the --spectum option https://gnpy.readthedocs.io/en/master/json.html#arbitrary-channel-definition (keyword is "specrum", see this example https://github.com/Telecominfraproject/oopt-gnpy/blob/master/gnpy/example-data/initial_spectrum1.json)

**Reply by SKZnljx** [reply]
Thank you very much for your response. 
I have another question: if I want to set a certain service to be unmodulated, can I directly set its baud rate to 0 or a very small number? Or is there another method?
